### PR TITLE
chore: update WebGL context docs

### PR DESCRIPTION
**Pull request by cc7gs**
Updates to ensure synchronisation with the English version

### CODE REVIEW

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 1]
```diff
@@ -50,6 +81,10 @@ slug: Web/API/WebGL_API/Tutorial/Adding_2D_content_to_a_WebGL_context
 
 **片段着色器**在顶点着色器处理完图形的顶点后，会被要绘制的每个图形的每个像素点调用一次。它的职责是确定像素的颜色，通过指定应用到像素的纹理元素（也就是图形纹理中的像素），获取纹理元素的颜色，然后将适当的光照应用于颜色。之后颜色存储在特殊变量 gl_FragColor 中，返回到 WebGL 层。该颜色将最终绘制到屏幕上图形对应像素的对应位置。
 
+在这种情况下，我们每次都会返回白色，因为我们只是在画一个白色的正方形，没有使用灯光。
+
+> **备注:** 添加下面代码到 `main()` 函数中:
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:86]: 同

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 2]
```diff
@@ -62,6 +97,8 @@ const fsSource = `
 
 现在我们已经定义了两个着色器，我们需要将它们传递给 WebGL，编译并将它们连接在一起。下面的代码通过调用 loadShader（），为着色器传递类型和来源，创建了两个着色器。然后创建一个附加着色器的程序，将它们连接在一起。如果编译或链接失败，代码将弹出 alert。
 
+> **备注:** 将下面两个函数添加到 "webgl-demo.js" 文件中:
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:100]: 同

```suggestion
> **备注：** 将下面两个函数添加到“webgl-demo.js”文件中:
```

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 3]
```diff
@@ -148,39 +189,37 @@ const programInfo = {
 
 在画正方形前，我们需要创建一个缓冲器来存储它的顶点。我们会用到名字为 initBuffers() 的函数。当我们了解到更多 WebGL 的高级概念时，这段代码会将有更多参数，变得更加复杂，并且用来创建更多的三维物体。
 
+> **备注:** 创建一个新文件叫 "init-buffers.js",并填写如下内容:
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:192]:
```suggestion
> **备注：** 创建一个新文件叫“init-buffers.js”，并填写如下内容：
```

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 4]
```diff
@@ -191,7 +230,7 @@ function initBuffers(gl) {
 
 当着色器和物体都创建好后，我们可以开始渲染这个场景了。因为我们这个例子不会产生动画，所以 drawScene() 方法非常简单。它还使用了几个工具函数，稍后我们会介绍。
 
-> **备注：** 你可能会得到这样一段错误报告：“mat4 is not defined”，意思是说你缺少`glmatrix`库。该库的 js 文件[gl-matrix.js](https://mdn.github.io/webgl-examples/tutorial/gl-matrix.js)可以从[这里](https://github.com/mdn/webgl-examples/issues/20)获得。
+> **备注:** 创建一个新文件叫 "init-buffers.js",并填写如下内容:
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:233]:
```suggestion
> **备注：** 创建一个新文件叫“draw-scene.js”，并填写如下内容:
```

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 5]
```diff
@@ -278,12 +317,55 @@ function drawScene(gl, programInfo, buffers) {
...
+export { drawScene };
 ```
 
 第一步，用背景色擦除画布，接着建立摄像机透视矩阵。设置 45 度的视图角度，并且设置一个适合实际图像的宽高比。指定在摄像机距离 0.1 到 100 单位长度的范围内的物体可见。
 
 接着加载特定位置，并把正方形放在距离摄像机 6 个单位的的位置。然后，我们绑定正方形的顶点缓冲到上下文，并配置好，再通过调用 {{domxref("WebGLRenderingContext.drawArrays()", "drawArrays()")}} 方法来画出对象。
 
+最后, 让我们引入 `initBuffers()` 和 `drawScene()`.
+
+> **备注:** 在 "webgl-demo.js" 文件头部添加如下代码:
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:351]: 标点

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 6]
```diff
@@ -294,6 +376,7 @@ function drawScene(gl, programInfo, buffers) {
 
 ## 相关资料
 
+- [Matrices](https://webglfundamentals.org/webgl/lessons/webgl-2d-matrices.html) WebGL基础知识
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:379]: 不应翻译 `WebGLFundamentals`

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 7]
```diff
@@ -294,6 +376,7 @@ function drawScene(gl, programInfo, buffers) {
...
 - [Matrices](http://mathworld.wolfram.com/Matrix.html) 线上数学百科全书
 - [Matrix](<http://en.wikipedia.org/wiki/Matrix_(mathematics)>) 维基百科
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:381]:
```suggestion
- 维基百科上的[矩阵](https://zh.wikipedia.org/wiki/矩阵) 
```

**Review by yin1999 — CHANGES_REQUESTED** [review]

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 8]
```diff
@@ -148,39 +189,37 @@ const programInfo = {
 
 在画正方形前，我们需要创建一个缓冲器来存储它的顶点。我们会用到名字为 initBuffers() 的函数。当我们了解到更多 WebGL 的高级概念时，这段代码会将有更多参数，变得更加复杂，并且用来创建更多的三维物体。
 
+> **备注：** 创建一个新文件叫 “init-buffers.js”，并填写如下内容：
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:192]:
```suggestion
> **备注：** 创建一个新文件叫“init-buffers.js”，并填写如下内容：
```

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 9]
```diff
@@ -191,7 +230,7 @@ function initBuffers(gl) {
 
 当着色器和物体都创建好后，我们可以开始渲染这个场景了。因为我们这个例子不会产生动画，所以 drawScene() 方法非常简单。它还使用了几个工具函数，稍后我们会介绍。
 
-> **备注：** 你可能会得到这样一段错误报告：“mat4 is not defined”，意思是说你缺少`glmatrix`库。该库的 js 文件[gl-matrix.js](https://mdn.github.io/webgl-examples/tutorial/gl-matrix.js)可以从[这里](https://github.com/mdn/webgl-examples/issues/20)获得。
+> **备注：** 创建一个新文件叫 “init-buffers.js”,并填写如下内容：
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:233]:
```suggestion
> **备注：** 创建一个新文件叫“draw-scene.js”，并填写如下内容：
```

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 10]
```diff
@@ -278,12 +317,55 @@ function drawScene(gl, programInfo, buffers) {
...
+export { drawScene };
 ```
 
 第一步，用背景色擦除画布，接着建立摄像机透视矩阵。设置 45 度的视图角度，并且设置一个适合实际图像的宽高比。指定在摄像机距离 0.1 到 100 单位长度的范围内的物体可见。
 
 接着加载特定位置，并把正方形放在距离摄像机 6 个单位的的位置。然后，我们绑定正方形的顶点缓冲到上下文，并配置好，再通过调用 {{domxref("WebGLRenderingContext.drawArrays()", "drawArrays()")}} 方法来画出对象。
 
+最后，让我们引入 `initBuffers()` 和 `drawScene()`。
+
+> **备注：** 在 “webgl-demo.js” 文件头部添加如下代码：
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:351]:
```suggestion
> **备注：** 在“webgl-demo.js”文件开头添加如下代码：
```

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 11]
```diff
@@ -294,7 +376,8 @@ function drawScene(gl, programInfo, buffers) {
 
 ## 相关资料
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:377]:
```suggestion
## 参见
```

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 12]
```diff
@@ -125,11 +162,15 @@ loadShader 函数将 WebGL 上下文，着色器类型和`源码`作为参数输
...
 我们可以像这样调用这段代码
 
+> **备注：** 添加下面代码到 `main()` 函数中：
+
 ```js
   const shaderProgram = initShaderProgram(gl, vsSource, fsSource);
 ```
 
-在创建着色器程序之后，我们需要查找 WebGL 返回分配的输入位置。在上述情况下，我们有一个属性和两个 uniforms。属性从缓冲区接收值。顶点着色器的每次迭代都从分配给该属性的缓冲区接收下一个值。uniforms 类似于 JavaScript 全局变量。它们在着色器的所有迭代中保持相同的值。由于属性和统一的位置是特定于单个着色器程序的，因此我们将它们存储在一起以使它们易于传递
+在创建着色器程序之后，我们需要查找 WebGL 返回分配的输入位置。在上述情况下，我们有一个属性和两个 [Uniforms](/en-US/docs/Web/API/WebGL_API/Data#uniforms) 。属性从缓冲区接收值。顶点着色器的每次迭代都从分配给该属性的缓冲区接收下一个值。uniforms 类似于 JavaScript 全局变量。它们在着色器的所有迭代中保持相同的值。由于属性和统一的位置是特定于单个着色器程序的，因此我们将它们存储在一起以使它们易于传递
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:171]:
```suggestion
在创建着色器程序之后，我们需要查找 WebGL 返回分配的输入位置。在上述情况下，我们有一个属性和两个 [Uniform](/zh-CN/docs/Web/API/WebGL_API/Data#uniforms) 。属性从缓冲区接收值。顶点着色器的每次迭代都从分配给该属性的缓冲区接收下一个值。uniform 类似于 JavaScript 全局变量。它们在着色器的所有迭代中保持相同的值。由于属性和统一的位置是特定于单个着色器程序的，因此我们将它们存储在一起以使它们易于传递
```

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 13]
```diff
@@ -21,14 +50,16 @@ slug: Web/API/WebGL_API/Tutorial/Adding_2D_content_to_a_WebGL_context
 
 #### 顶点着色器
 
-每次渲染一个形状时，顶点着色器会在形状中的每个顶点运行。它的工作是将输入顶点从原始坐标系转换到 WebGL 使用的缩放空间 (**clipspace**) 坐标系，其中每个轴的坐标范围从 -1.0 到 1.0，并且不考虑纵横比，实际尺寸或任何其他因素。
+每次渲染一个形状时，顶点着色器会在形状中的每个顶点运行。它的工作是将输入顶点从原始坐标系转换到 WebGL 使用的[**裁剪空间**](/zh-CN/docs/Web/API/WebGL_API/WebGL_model_view_projection#裁剪空间)**) 坐标系，其中每个轴的坐标范围从 -1.0 到 1.0，并且不考虑纵横比，实际尺寸或任何其他因素。
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:53]:
```suggestion
每次渲染一个形状时，顶点着色器会在形状中的每个顶点运行。它的工作是将输入顶点从原始坐标系转换到 WebGL 使用的[**裁剪空间**](/zh-CN/docs/Web/API/WebGL_API/WebGL_model_view_projection#裁剪空间)) 坐标系，其中每个轴的坐标范围从 -1.0 到 1.0，并且不考虑纵横比，实际尺寸或任何其他因素。
```

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 14]
```diff
@@ -7,6 +7,35 @@ slug: Web/API/WebGL_API/Tutorial/Adding_2D_content_to_a_WebGL_context
 
 一旦创建 WebGL 上下文创建成功，你就可以在这个上下文里渲染画图了。而对我们而言最简单的事，莫过于绘制一个没有纹理的 2D 图形了。那就让我们从画出一个正方形开始吧。
 
+本项目的完整源代码 [可在 GitHub 上获得](https://github.com/mdn/dom-examples/tree/main/webgl-examples/tutorial/sample2)。
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:10]:
```suggestion
本项目的完整源代码[可在 GitHub 上获得](https://github.com/mdn/dom-examples/tree/main/webgl-examples/tutorial/sample2)。
```

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 15]
```diff
@@ -7,6 +7,35 @@ slug: Web/API/WebGL_API/Tutorial/Adding_2D_content_to_a_WebGL_context
...
+
+## 引入 glMatrix 库
+
+该项目使用了 [glMatrix](https://glmatrix.net/) 库来执行其矩阵操作，因此需要引入它。本次示例通过 CDN 形式引入使用。
+
+> **备注：** 更新 “index.html” 文件添加 CDN 脚本：
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:16]:
```suggestion
> **备注：** 更新“index.html”文件，让它看起来像这样：
```

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 16]
```diff
@@ -62,6 +97,8 @@ const fsSource = `
 
 现在我们已经定义了两个着色器，我们需要将它们传递给 WebGL，编译并将它们连接在一起。下面的代码通过调用 loadShader（），为着色器传递类型和来源，创建了两个着色器。然后创建一个附加着色器的程序，将它们连接在一起。如果编译或链接失败，代码将弹出 alert。
 
+> **备注：** 将下面两个函数添加到 “webgl-demo.js” 文件中：
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:100]:
```suggestion
> **备注：** 将下面两个函数添加到“webgl-demo.js”文件中：
```

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 17]
```diff
@@ -62,6 +97,8 @@ const fsSource = `
 
 现在我们已经定义了两个着色器，我们需要将它们传递给 WebGL，编译并将它们连接在一起。下面的代码通过调用 loadShader（），为着色器传递类型和来源，创建了两个着色器。然后创建一个附加着色器的程序，将它们连接在一起。如果编译或链接失败，代码将弹出 alert。
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:98]:
```suggestion
现在我们已经定义了两个着色器，我们需要将它们传递给 WebGL，编译并将它们连接在一起。下面的代码通过调用 `loadShader()`，为着色器传递类型和来源，创建了两个着色器。然后创建一个附加着色器的程序，将它们连接在一起。如果编译或链接失败，代码将弹出 alert。
```

**Review by yin1999 — APPROVED** [review]
Thanks. Welcome aboard :tada:

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 18]
```diff
@@ -21,14 +50,16 @@ slug: Web/API/WebGL_API/Tutorial/Adding_2D_content_to_a_WebGL_context
 
 #### 顶点着色器
 
-每次渲染一个形状时，顶点着色器会在形状中的每个顶点运行。它的工作是将输入顶点从原始坐标系转换到 WebGL 使用的缩放空间 (**clipspace**) 坐标系，其中每个轴的坐标范围从 -1.0 到 1.0，并且不考虑纵横比，实际尺寸或任何其他因素。
+每次渲染一个形状时，顶点着色器会在形状中的每个顶点运行。它的工作是将输入顶点从原始坐标系转换到 WebGL 使用的[**裁剪空间**](/zh-CN/docs/Web/API/WebGL_API/WebGL_model_view_projection#裁剪空间)) 坐标系，其中每个轴的坐标范围从 -1.0 到 1.0，并且不考虑纵横比，实际尺寸或任何其他因素。
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:53]:
```suggestion
每次渲染一个形状时，顶点着色器会在形状中的每个顶点运行。它的工作是将输入顶点从原始坐标系转换到 WebGL 使用的[**裁剪空间**](/zh-CN/docs/Web/API/WebGL_API/WebGL_model_view_projection#裁剪空间)坐标系，其中每个轴的坐标范围从 -1.0 到 1.0，并且不考虑纵横比，实际尺寸或任何其他因素。
```

File: files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md [hunk 19]
```diff
@@ -21,14 +50,16 @@ slug: Web/API/WebGL_API/Tutorial/Adding_2D_content_to_a_WebGL_context
...
 
-每次渲染一个形状时，顶点着色器会在形状中的每个顶点运行。它的工作是将输入顶点从原始坐标系转换到 WebGL 使用的缩放空间 (**clipspace**) 坐标系，其中每个轴的坐标范围从 -1.0 到 1.0，并且不考虑纵横比，实际尺寸或任何其他因素。
+每次渲染一个形状时，顶点着色器会在形状中的每个顶点运行。它的工作是将输入顶点从原始坐标系转换到 WebGL 使用的[**裁剪空间**](/zh-CN/docs/Web/API/WebGL_API/WebGL_model_view_projection#裁剪空间)坐标系，其中每个轴的坐标范围从 -1.0 到 1.0，并且不考虑纵横比，实际尺寸或任何其他因素。
 
 顶点着色器需要对顶点坐标进行必要的转换，在每个顶点基础上进行其他调整或计算，然后通过将其保存在由 GLSL 提供的特殊变量（我们称为 gl_Position）中来返回变换后的顶点
 
-顶点着色器根据需要，也可以完成其他工作。例如，决定哪个包含 [texel](<https://zh.wikipedia.org/wiki/texel_(graphics)>) 面部纹理的坐标，可以应用于顶点；通过法线来确定应用到顶点的光照因子等。然后将这些信息存储在[变量（varyings)](/zh-CN/docs/Web/API/WebGL_API/Data#varyings)或[属性 (attributes)](/zh-CN/docs/Web/API/WebGL_API/Data#Attributes)属性中，以便与片段着色器共享
+顶点着色器根据需要，也可以完成其他工作。例如，决定哪个包含 {{Glossary("texel")}} 面部纹理的坐标，可以应用于顶点；通过法线来确定应用到顶点的光照因子等。然后将这些信息存储在[变量（varyings)](/zh-CN/docs/Web/API/WebGL_API/Data#varyings)或[属性 (attributes)](/zh-CN/docs/Web/API/WebGL_API/Data#Attributes)属性中，以便与片段着色器共享
 
 以下的顶点着色器接收一个我们定义的属性（aVertexPosition）的顶点位置值。之后这个值与两个 4x4 的矩阵（uProjectionMatrix 和 uModelMatrix）相乘; 乘积赋值给 gl_Position。有关投影和其他矩阵的更多信息，[在这里您可能可以找到有帮助的文章](https://webglfundamentals.org/webgl/lessons/webgl-3d-perspective.html).。
```
Comment (yin1999) [inline files/zh-cn/web/api/webgl_api/tutorial/adding_2d_content_to_a_webgl_context/index.md:59]:
```suggestion
以下的顶点着色器接收一个我们定义的属性（aVertexPosition）的顶点位置值。之后这个值与两个 4x4 的矩阵（uProjectionMatrix 和 uModelMatrix）相乘; 乘积赋值给 gl_Position。有关投影和其他矩阵的更多信息，[在这里您可能可以找到有帮助的文章](https://webglfundamentals.org/webgl/lessons/webgl-3d-perspective.html)。
```

**Comment by cc7gs** [comment]
> 

Thanks  🙏

**Pull request merged**